### PR TITLE
Merging updating tokens script to master

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by jakirkham** [comment]
Does this update all branches of all feedstocks or only `master`?

**Comment by isuruf** [comment]
@jakirkham, can we merge this and raise an issue about it?

File: .travis.yml [hunk 1]
```diff
@@ -23,11 +23,21 @@ env:
         # Add the TRAVIS_TOKEN environment variable for the "conda-forge" Travis user.
         - secure: "BBJrw5OcEG4zEUxjXt2txNoz+o7ilTb0e8Tj1626USKqi6HAWOdSFHW7PLYR6vyefWsHnGLAeYMRr9UB56O2oDwS7jgClBKbyut2+Q+8SQBwEtNCMHdU996IzSuDP4AM3h7DLjTsuinWhwpqLu72z2ZjE9OMz4q03SyyiBOecWU1btYbp/u3bpNzyhb2u0HMkRUnS8MxPUa8aHZBKp/JpLrGQmIT0LcjLNp4/jTZp+cs+J//02hAhJHV3vMmyefVvRZayu3xKzutpb3qIga0JAP3d0j2FWXRHVTW6Ke5H4u/4NU7iTpdRjpymnzH7H2kfOrNzgCJ23NfNhScDJDc5URvdXlstgkC4lDkd7BbSVgKKaVy2vVcMkz98qbBnhi0akMHNZXPLvT9tec9AsAX5/pdNaa1rH/iRhjWRfTCnqODX+RTwMhA1/3rkmhzI/8JFCcaZet8+lA2LtOf95jGmwcRwPlNFX0Jdd1QBk2nUAUpCePWIdg7GRxYuO7a4QCbhc6bSyB/WrCQZ1oEmtwGSRqxpMKvQmjxJObg40MgP0DaquGov30A4QLoEdvXCnM7w3QyIr5L/BrLlnCy8LesAZIBn2l5LAHrBFGh9t/lCK9APlUjD9dgT6eCYOSbfdBghKCGsajAsM/fwsPvcjOmpFCLf+UZTer36B57ccHrYdU="
 
+    #matrix:
+    #    - TEST_SPLIT=0/5
+    #    - TEST_SPLIT=1/5
+    #    - TEST_SPLIT=2/5
+    #    - TEST_SPLIT=3/5
+    #    - TEST_SPLIT=4/5
```
Comment (jakirkham) [inline .travis.yml]: Is this suppose to be commented?

Comment (isuruf) [reply .travis.yml]: yes, when we want to run this script it has to be uncommented

File: .CI/update_all_tokens [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    curl -L https://repo.continuum.io/miniconda/Miniconda2-latest-MacOSX-x86_64.sh > ~/miniconda.sh
+    bash ~/miniconda.sh -b -p ~/miniconda
+    source ~/miniconda/bin/activate root
+
+    # Configure conda.
+    echo ""
+    echo "Configuring conda."
+    conda config --set show_channel_urls true
+    conda config --add channels conda-forge
+    conda install --yes --quiet git=2.12.2
```
Comment (jakirkham) [inline .CI/update_all_tokens]: We should re-evaluate this pinning at some point, but it doesn't need to happen in this PR.

File: .CI/update_all_tokens.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+if 'APPVEYOR_TOKEN' in os.environ:
+    write_token('appveyor', os.environ['APPVEYOR_TOKEN'])
+if 'CIRCLE_TOKEN' in os.environ:
+    write_token('circle', os.environ['CIRCLE_TOKEN'])
+if 'GH_TOKEN' in os.environ:
+    write_token('github', os.environ['GH_TOKEN'])
+
+from conda_smithy.ci_register import (add_token_to_circle, travis_headers,
+    travis_encrypt_binstar_token, appveyor_encrypt_binstar_token,
+    travis_token_update_conda_forge_config)
```
Comment (jakirkham) [inline .CI/update_all_tokens.py]: It's weird that with AppVeyor we join these into one step (encrypting the token and adding it to the config file), but Travis CI we do not. Nothing we can do about it here for now, but maybe we should pick one of these approaches.

File: .CI/update_all_tokens (hunk 2 above)
Comment (isuruf) [reply .CI/update_all_tokens]: I think I copied this script from a repo where there was an issue with the new git. Can't remember which one

File: .travis.yml (hunk 1 above)
Comment (jakirkham) [reply .travis.yml]: Would it make sense to leave this uncommented so we can easily run it? That or should we add a comment above explaining this should be uncommented to run the script?

Comment (isuruf) [reply .travis.yml]: Leaving this uncommented would add 5 builds to each PR run here.

File: .CI/update_all_tokens (hunk 2 above)
Comment (jakirkham) [reply .CI/update_all_tokens]: Yep, I think the pinning is my doing actually. Though I think it is from over a year ago and is probably not needed any more.

File: .travis.yml (hunk 1 above)
Comment (jakirkham) [reply .travis.yml]: Sorry missed this was going in `master` for some reason even though it is clearly in the title. :man_facepalming:

Comment (jakirkham) [reply .travis.yml]: I guess just a comment above should suffice.

File: .CI/update_all_tokens (hunk 2 above)
Comment (isuruf) [reply .CI/update_all_tokens]: Nope. it was me, https://github.com/conda-forge/staged-recipes/commit/3c7a57397ff764d71b285b97aba951c2d8db7027

**Comment by jakirkham** [comment]
Does this still need to be merged, @isuruf?

**Comment by isuruf** [comment]
@conda-forge-admin, please lint

**Comment by isuruf** [comment]
@conda-forge-admin, please lint

**Pull request closed without merging**